src/api/app/models/workflow/step.rb

```diff
@@ -150,8 +150,8 @@ def report_to_scm(workflow_filters)
 
   # Only used in LinkPackageStep and BranchPackageStep.
   def validate_source_project_and_package_name
-    errors.add(:base, "invalid source project '#{source_project_name}'") unless Project.valid_name?(source_project_name)
-    errors.add(:base, "invalid source package '#{source_package_name}'") unless Package.valid_name?(source_package_name)
+    errors.add(:base, "invalid source project '#{source_project_name}'") if step_instructions[:source_project] && !Project.valid_name?(source_project_name)
+    errors.add(:base, "invalid source package '#{source_package_name}'") if step_instructions[:source_package] && !Package.valid_name?(source_package_name)
```

Comment:
A small nitpick for consistency... The code for `target_project` could be in the same format as this line.

Reply:
Done.